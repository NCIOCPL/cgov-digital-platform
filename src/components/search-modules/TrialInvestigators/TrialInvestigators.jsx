import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Fieldset, Autocomplete } from '../../atomic';
import { searchTrialInvestigators } from '../../../store/actions';
import { matchItemToTerm, sortItems } from '../../../utilities/utilities';

const TrialInvestigators = ({ handleUpdate }) => {
  const dispatch = useDispatch();
  const [tiName, setTiName] = useState({ value: '' });

  //store vals
  const { tis = [] } = useSelector(store => store.cache);

  useEffect(() => {
    handleUpdate('inv', tiName);
  }, [tiName, handleUpdate]);

  useEffect(() => {
    if (tiName.value.length > 2) {
      dispatch(searchTrialInvestigators({ searchText: tiName.value }));
    }
  }, [tiName, dispatch]);


  return (
    <Fieldset
      id="trialInvestigators"
      legend="Trial Investigators"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialinvestigators"
    >
      <Autocomplete
        id="inv"
        label="Trial investigators"
        labelHidden
        inputHelpText="Search by Trial Investigators"
        value={tiName.value}
        inputProps={{ id: 'in', placeholder: 'Investigator name' }}
        wrapperStyle={{ position: 'relative', display: 'inline-block' }}
        items={tis}
        getItemValue={item => item.term}
        shouldItemRender={matchItemToTerm}
        sortItems={sortItems}
        onChange={(event, value) => setTiName({ value })}
        onSelect={value => setTiName({ value })}
        renderMenu={children => (
          <div className="cts-autocomplete__menu --trialInvestigators">
            {(tiName.value.length > 2 )
                ? (tis.length)
                    ? (children)
                    : <div className="cts-autocomplete__menu-item">No results found</div>
                : <div className="cts-autocomplete__menu-item">Please enter 3 or more characters</div>
            }
          </div>
        )}
        renderItem={(item, isHighlighted) => (
          <div
            className={`cts-autocomplete__menu-item ${
              isHighlighted ? 'highlighted' : ''
            }`}
            key={item.term_key}
          >
            {item.term}
          </div>
        )}
      />
    </Fieldset>
  );
};

export default TrialInvestigators;
