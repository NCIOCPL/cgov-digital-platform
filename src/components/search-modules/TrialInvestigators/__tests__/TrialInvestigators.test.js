import React from 'react';
import { render } from 'enzyme';
import TrialInvestigators from '../TrialInvestigators';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <TrialInvestigators {...props} />
  );
  return component;
};

describe('TrialInvestigators', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
