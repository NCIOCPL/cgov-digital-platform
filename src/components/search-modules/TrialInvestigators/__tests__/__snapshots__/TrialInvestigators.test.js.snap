// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialInvestigators Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--trialinvestigator"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Trial Investigators
    </span>
  </legend>
  <a
    aria-label="Help"
    class="text-icon-help"
    href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialinvestigators"
    rel="noopener noreferrer"
    target="_blank"
  >
    ?
  </a>
  <div
    class="cts-fieldset__body"
  >
    <div
      class=""
    >
      <label
        class="cts-label"
        for="in"
        id="in-label"
      >
        Search by Trial Investigators.
      </label>
      <input
        aria-labelledby="in-label"
        aria-required="false"
        class="cts-input  "
        id="in"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
